test/e2e/hsts_policy_test.go

```
@@ -0,0 +1,207 @@
+		t.Fatalf("failed to get ingresscontroller service: %v", err)
+	}
+
+	// Create some pods in the deployment namespace
+	echoPod := buildEchoPod("hsts-policy-echo", deployment.Namespace)
+	if err := kclient.Create(context.TODO(), echoPod); err != nil {
+		t.Fatalf("failed to create pod %s/%s: %v", echoPod.Namespace, echoPod.Name, err)
+	}
+	defer func() {
+		if err := kclient.Delete(context.TODO(), echoPod); err != nil {
+			t.Fatalf("failed to delete pod %s/%s: %v", echoPod.Namespace, echoPod.Name, err)
+		}
+	}()
+
+	echoService := buildEchoService(echoPod.Name, echoPod.Namespace, echoPod.ObjectMeta.Labels)
+	if err := kclient.Create(context.TODO(), echoService); err != nil {
+		t.Fatalf("failed to create service %s/%s: %v", echoService.Namespace, echoService.Name, err)
+	}
+	defer func() {
+		if err := kclient.Delete(context.TODO(), echoService); err != nil {
+			t.Fatalf("failed to delete service %s/%s: %v", echoService.Namespace, echoService.Name, err)
+		}
+	}()
+
+	// Create a namespace for the new routes
+	ns := &corev1.Namespace{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "hsts-policy-namespace",
+		},
+	}
+	if err := kclient.Create(context.TODO(), ns); err != nil {
+		t.Fatalf("failed to create namespace: %v", err)
+	}
+	defer func() {
+		// this will also clean up all the routes
+		if err := kclient.Delete(context.TODO(), ns); err != nil {
+			t.Fatalf("failed to delete test namespace %s: %v", ns.Name, err)
+		}
+	}()
+
+	// Create routes to test performance aspect
+	numRoutes := 1000
+	t.Logf("creating %d routes at %s...", numRoutes, time.Now().Format(time.StampMilli))
+	for i := 0; i < numRoutes; i++ {
+		r := buildRouteWithHSTS(echoPod.Name+strconv.Itoa(i), ns.Name, echoService.Name, domain, "max-age=99999;preload;includesubdomains")
+		if err := kclient.Create(context.TODO(), r); err != nil {
+			t.Logf("failed to create route %s/%s: %v", r.Namespace, r.Name, err)
```

Shouldn't this fail early?

---

Removed the 1000 routes tests altogether.